src/FSharpVSPowerTools/Commands/PrintfSpecifiersUsageTaggerProvider.cs

```diff
@@ -0,0 +1,129 @@
+using System.Windows.Media;
+using Microsoft.VisualStudio.Text.Classification;
+
+namespace FSharpVSPowerTools
+{
+    [Export(typeof(IViewTaggerProvider))]
+    [ContentType("F#")]
+    [TagType(typeof(PrintfSpecifiersUsageTag))]
+    public class PrintfSpecifiersUsageTaggerProvider : IViewTaggerProvider, IDisposable
+    {
+        readonly IServiceProvider _serviceProvider;
+        readonly ITextDocumentFactoryService _textDocumentFactoryService;
+        readonly ProjectFactory _projectFactory;
+        readonly VSLanguageService _fsharpVsLanguageService;
+        readonly ShellEventListener _shellEventListener;
+        readonly PrintfColorManager _printfColorManager;
+
+        [ImportingConstructor]
+        public PrintfSpecifiersUsageTaggerProvider(
+            [Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider,
+            ITextDocumentFactoryService textDocumentFactoryService,
+            ProjectFactory projectFactory,
+            VSLanguageService fsharpVsLanguageService,
+            ShellEventListener shellEventListener,
+            PrintfColorManager printfColorManager)
+        {
+            _serviceProvider = serviceProvider;
+            _textDocumentFactoryService = textDocumentFactoryService;
+            _projectFactory = projectFactory;
+            _fsharpVsLanguageService = fsharpVsLanguageService;
+            _shellEventListener = shellEventListener;
+            _printfColorManager = printfColorManager;
+            _shellEventListener.ThemeChanged += UpdateTheme;
+        }
+
+        void UpdateTheme(object sender, EventArgs e)
+        {
+            _printfColorManager.UpdateColors();
+        }
+
+        public ITagger<T> CreateTagger<T>(ITextView textView, ITextBuffer buffer) where T : ITag
+        {
+            // Only provide highlighting on the top-level buffer
+            if (textView.TextBuffer != buffer) return null;
+
+            var generalOptions = Setting.getGeneralOptions(_serviceProvider);
+            if (generalOptions == null || !generalOptions.HighlightPrintfUsageEnabled) return null;
+
+            ITextDocument doc;
+            if (_textDocumentFactoryService.TryGetTextDocument(buffer, out doc))
+            {
+                return buffer.Properties.GetOrCreateSingletonProperty(
+                    () => new PrintfSpecifiersUsageTagger(doc, textView, _fsharpVsLanguageService, _serviceProvider, _projectFactory)) as ITagger<T>;
+            }
+
+            return null;
+        }
+
+        public void Dispose()
+        {
+            _shellEventListener.ThemeChanged -= UpdateTheme;
+        }
+    }
+
+    [Export]
+    public class PrintfColorManager
+    {
+        static readonly Color LightThemeColor = Color.FromRgb(245, 222, 179);
+        static readonly Color DarkThemeColor = Color.FromRgb(0, 77, 77);
+        VisualStudioTheme lastTheme = VisualStudioTheme.Unknown;
+
+        [Import]
+        ThemeManager themeManager = null;
```

Comment:
You can use `ImportingConstructor` here. It's better and more consistent with the rest of the project.